src/net_processing.cpp

```diff
@@ -3358,7 +3375,13 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
             // requests for it.
             m_txrequest.ForgetTxHash(tx.GetHash());
             m_txrequest.ForgetTxHash(tx.GetWitnessHash());
-            RelayTransaction(tx.GetHash(), tx.GetWitnessHash(), m_connman);
+
+            // Flood those transactions which were received either via flooding, or inbound reconciliation,
+            // but NOT via outbound reconciliation. Flooding then is mainly used for initial propagation
+            // of new transactions across a network of reachable nodes quickly.
```

Comment:
>IIUC, this comment correctly, you mean that a transaction discovered through an accepted-reconciliation will be marked for reconciliation for its future announcements, no matter the peer type (strategic outbound/non-strategic outbound/inbound).

Correct if you mean accepted=outbound (see confusion below)

> I don't get the rational of such approach, propagation of such transactions could be accelerated if we were flooding them for our outbound peers.

Consider a non-reachable peer with 8 reconciliation-supporting outbound peers. They will be all marked strategic. If we do what you say, we will flood transactions to all of them. I suggest limiting it: since a non-reachable node received it from an outbound reconciliation (that's the only way it could have received a tx), it won't flood them to anyone.

>Also you may consider "inbound reconciliation" -> "initiated reconciliation", "outbound reconciliation" -> "accepted reconciliation". Even if reconciliation roles are decided in function of our peer selection, they should be clearly dissociated as orthogonal.

ACK, although I think "inbound" is not initiated, but it's accepted.